File: gracie.lee/src/main/java/homework/chapter_6/homework2/Member.java
package homework.chapter_6.homework2;

import jakarta.persistence.*;

@Entity(name = "Hw2Member")
@Table(name = "member_hw2")
public class Member {

    @Id
    @GeneratedValue
    @Column(name = "member_id")
    private Long id;

    private String name;

    @OneToOne  // ← fetch = FetchType.LAZY 추가해서 지연 로딩 확인!
    @JoinColumn(name = "locker_id")
    private Locker locker;

    protected Member() {
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Locker getLocker() {
        return locker;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setLocker(Locker locker) {
        this.locker = locker;
    }
}

File: gracie.lee/src/main/kotlin/homework/chapter_6/homework2/Member.kt
package homework.chapter_6.homework2

import jakarta.persistence.*

@Entity(name = "Hw2MemberKt")
@Table(name = "member_hw2_kt")
class Member(
    @Id
    @GeneratedValue
    @Column(name = "member_id")
    val id: Long? = null,

    var name: String? = null,

    @OneToOne  // <- fetch = FetchType.LAZY 추가해서 지연 로딩 확인!
    @JoinColumn(name = "locker_id")
    var locker: Locker? = null
)
